Source/WebKit/Platform/IPC/StreamServerConnection.cpp

```diff
@@ -174,13 +175,13 @@ StreamServerConnection::DispatchResult StreamServerConnection::dispatchStreamMes
         }
         if (!currentReceiver) {
             // Valid scenario is when receiver has been removed, but there are messages for it in the buffer.
-            // FIXME: Since we do not have a receiver, we don't know how to decode the message.
-            // This means we must timeout every receiver in the stream connection.
-            // Currently we assert that the receivers are empty, as we only have up to one receiver in
-            // a stream connection until possibility of skipping is implemented properly.
-            Locker locker { m_receiversLock };
-            ASSERT(m_receivers.isEmpty());
-            return DispatchResult::HasNoMessages;
+            if (skipStreamMessage(decoder)) {
```

Comment:
Fixing the assertion bug in WebGPU doesn't sound like it'd need the skip implementation..

The idea is that if there is a malformed message in the IPC data stream, the whole data stream is invalid. E.g. you cannot calculate where  the next message is in a invalid data stream.

It is a sender error to send a malformed message, and as such the only fix should be to fix the sender to not send the malformed message.

Also adding nontrivial features to IPC could make use of the unit tests. It'd be good to then maybe consider implementing non-trivial features as separate patches to the other fixes, to ease the review and discussion.

Reply:
Discussed with @achristensen07 and found a four line change to fix the crash instead